torch/sparse/semi_structured.py

```diff
@@ -0,0 +1,362 @@
+    Additionally, your tensor must be a positive multiple of a block size given the dtype
+
+    - torch.float16  (r, c) must be >= and a multiple of 64
+    - torch.int8     (r, c) must be >= and a multiple of 128
+
+    Args:
+        original_tensor (Tensor): the dense tensor to convert
+        mask (Optional BoolTensor): boolean mask to apply to the original tensor
+        transposed (bool, optional): whether the dense tensor is transposed
+
+    Returns:
+        SparseSemiStructuredTensor: A sparse semi-structured tensor created from the given original_tensor and mask
+
+    Raises:
+        NotImplementedError: If ``mask=None``, as we currently do not support inferring a mask from the dense tensor.
+        RuntimeError: If original_tensor is not a supported dtype, dim, shape, or device.
+
+    Example:
+        >>> # xdoctest: +REQUIRES(env:TORCH_DOCTEST_CUDA)
+        >>> A = torch.Tensor([0, 0, 1, 1]).tile((128, 32)).half().cuda()
+        tensor([[0., 0., 1.,  ..., 0., 1., 1.],
+                [0., 0., 1.,  ..., 0., 1., 1.],
+                [0., 0., 1.,  ..., 0., 1., 1.],
+                ...,
+                [0., 0., 1.,  ..., 0., 1., 1.],
+                [0., 0., 1.,  ..., 0., 1., 1.],
+                [0., 0., 1.,  ..., 0., 1., 1.]], device='cuda:0', dtype=torch.float16)
+        >>> A_sparse = to_sparse_semi_structured(A, mask=A.bool())
+        SparseSemiStructuredTensor(shape=torch.Size([128, 128]), transposed=False, values=tensor([[1., 1., 1.,  ..., 1., 1., 1.],
+                [1., 1., 1.,  ..., 1., 1., 1.],
+                [1., 1., 1.,  ..., 1., 1., 1.],
+                ...,
+                [1., 1., 1.,  ..., 1., 1., 1.],
+                [1., 1., 1.,  ..., 1., 1., 1.],
+                [1., 1., 1.,  ..., 1., 1., 1.]], device='cuda:0', dtype=torch.float16),
+            metadata=tensor([[-4370, -4370, -4370,  ..., -4370, -4370, -4370],
+                [-4370, -4370, -4370,  ..., -4370, -4370, -4370],
+                [-4370, -4370, -4370,  ..., -4370, -4370, -4370],
+                ...,
+                [-4370, -4370, -4370,  ..., -4370, -4370, -4370],
+                [-4370, -4370, -4370,  ..., -4370, -4370, -4370],
+                [-4370, -4370, -4370,  ..., -4370, -4370, -4370]], device='cuda:0',
+       dtype=torch.int16))
+    """
+    warnings.warn(
+        (
+            "The PyTorch API of SparseSemiStructuredTensor is in prototype stage "
+            "and will change in the near future. Please open a Github issue "
+            "for features requests and see our documentation on the torch.sparse "
+            "module for further information about the project."
+        ),
+        UserWarning,
+    )
+
+    # check if mask passed in
+    if mask is None:
+        raise NotImplementedError(
+            (
+                "Creating mask from dense tensor is currently not supported! "
+                "You must pass in a mask to to_sparse_semi_structured, currently mask=None."
```

Comment:
This is quite unclear. Is this optional to future proof the API?
If so, I would just say that mask is mandatory in the error and maybe that users should open an issue if they want optional mask support added.